### DISCUSSION TITLE
Writing multi-select to database rows

### DISCUSSION THREAD

**Discussion by snvgglebear** [Budibase/budibase#6070] [Help]
I would like to take each item selected in a multi-select list and write each selected item to a separate database row. 
(I could also do what I want using selected rows from a table component, but I do not know how to do that either. )

**Comment by melohagan** [comment]
Hi @gondilon 

If you know the maximum number of options, and it isn't a huge number, then you could try this setup:
![Screenshot 2022-05-26 at 14 45 48](https://user-images.githubusercontent.com/101575380/170500227-5def720e-788a-4498-99f8-385a53ba5238.png)

#### Stop if
![Screenshot 2022-05-26 at 14 46 17](https://user-images.githubusercontent.com/101575380/170500316-44f9a896-7479-4a2d-b852-d4a908d49326.png)

For each *Stop if* action, increment the number by one.

#### Save Row
![Screenshot 2022-05-26 at 14 48 57](https://user-images.githubusercontent.com/101575380/170500813-7aefe5da-e0ea-40f9-8a4f-9874df7cd396.png)

Add the column you want to save the selected multi-select value to, and enter this javascript:
`return $("Form.Fields.Multi")[0];`
As before, increment the index by one for each following *Save Row* action.

---
I believe there are plans in the future to add looping to design actions which would make this less cumbersome, similar to how they are in automations: https://docs.budibase.com/docs/automation-actions#looping

**Reply by snvgglebear** [reply]
Can I not use the Foreach syntax inside a binding for this?

**Reply by melohagan** [reply]
Hey @gondilon unfortunately not because you need to trigger the **Save Row** action on each iteration, which currently isn't possible.

**Reply by snvgglebear** [reply]
@melohagan I do not see automation looping in my version of Budibase. (I am running version 1.0.110). Do I need to install a beta version to get it?

**Reply by melohagan** [reply]
The latest version is v1.0.158 I think. There should be a blue refresh icon you can click:

![Screenshot 2022-05-26 at 15 48 29](https://user-images.githubusercontent.com/101575380/170513027-a7d8fc44-08f6-4c85-8ed9-b7085fc1fd4e.png)

**Reply by snvgglebear** [reply]
I tried that, and it says 1.0.110 is the latest version available.

**Reply by snvgglebear** [reply]
Is there a way to manually update?

**Reply by melohagan** [reply]
Are you self hosted? It may be worth pulling down the latest docker-compose file and doing a `docker-compose up`.
You can also update through the budibase cli: https://docs.budibase.com/docs/budibase-cli-setup

`budi hosting --update`

**Reply by NicolasFabre81** [reply]
Is there now a way to achieve it smarter ? Maybe with automation ?
I also need to delete or update the rows if the user uncheck the boxes...

**Reply by melohagan** [reply]
Yeah I think since this answer was given, automation looping was introduced which may make things a bit easier! https://docs.budibase.com/docs/looping
